cmd/snap/cmd_connections.go

```diff
@@ -78,11 +78,12 @@ func endpoint(snap, name string) string {
 }
 
 type connection struct {
-	slot          string
-	plug          string
-	interfaceName string
-	manual        bool
-	gadget        bool
+	slot              string
+	plug              string
+	interfaceName     string
+	manual            bool
+	gadget            bool
+	definingAttrValue string
```

Comment:
Bikeshed time: `principleAttrValue` - using the word `defining` is correct but can be misread as `definition of` - especially in a variable name with camel-case.

Reply:
it's also slightly misleading to call *AttrValue, given that is the value wrapped in [] actually

Reply:
Perhaps this ought to be `*string` and should be rendered as `[%s]` whenever non-nil?

Reply:
I don't mind too much if this containts the [] or not, in the end this is the command impl. Let's just move it closer to interfaceName,  my  suggestion would be:

    interfaceName           string
    interfaceDeterminant string

Reply:
From this meaning for determinant:  "A determining factor; an element that determines the nature of something."